entrypoint.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+set -euo pipefail
+
+test -z "${STATA_LICENSE:-}" && { echo "No STATA_LICENSE environment variable found"; exit 1; }
+echo "$STATA_LICENSE" > /usr/local/stata/stata.lic
+
+script=$1
+shift
+
+test -f "$script" || { echo "$script does not exist"; exit 1; }
+
+# stata is super odd in its cli interactions and behaviour. So we wrap up the
+# actual script
+
+tmp=$(mktemp)
+wrapper=wrapper.$script
```

Comment:
arg, good point.

So, I didn't want to use temp, as that made for ugly logs for users to see (I want to link the wrapper name to their name as much as possible). It's easy enough to fix though, with `${script%.do}.wrapper.do`

Reply:
Bash string transformations will never cease to amaze me!

I agree it makes sense to try to keep the name sensible-looking, I just wonder if there's any way to do that while keeping it out of `/workspace`. The `trap .. rm` solution is neat but there's still something a bit yucky about writing temporary stuff as root in the working directory. What about something like replacing the `workspace` path segment with `wrapped` so you'd get e.g. `/wrapped/analysis/script.do`? Not saying that's a great solution either but feels worth thinking through the solution space a bit further.

Reply:
I think we need to be explicit about the fact their *is* a wrapper, but obvious *what* its wrapping.

Which for me means the basename not being the same as the scriptname (e.g. I expect most people would not notice `/wrapped/` above and they'd just read `/workspace/` there, which potential confusion),,

e.g. if their action is `stata analysis/model.do`, the log files will talk about `analysis/model.wrapper.do`, i.e same path prefix.

Also, I'm wary of putting it in a different directory, in case there are some odd CWD semantics in stata.

We could do `analysis/wrapper-model.do` or something similar, but stata seems to require a `.do` extension.